source/generate_source_file.py

```diff
@@ -0,0 +1,209 @@
+# Copyright (C) 2020 and later: Google, Inc.
+
+"""This purpose of this script is to create source .txt files. For statistical
+analysis on distribution of radicals and strokes, see Notebook script
+'Radical-stroke_Index_Analysis.ipynb'.
+Usage:
+    """
+
+import os
+import pandas as pd
+import random
+
+
+def main():
+    # Download Unihan meta data for radical-stroke analysis
+    os.system(' mkdir Unihan')
+    os.system(' curl -O http://unicode.org/Public/UCD/latest/ucd/Unihan.zip')
+    os.system(' apt-get -y install unzip')
+    os.system(' unzip Unihan.zip -d Unihan/')
+    os.system(' rm Unihan.zip')
+
+    # Make sure required package is installed
+    os.system(' python3 -m pip install pandas ')
+
+    # Create intermediate txt file that includes meta data
+    with open('Unihan/Unihan_RadicalStrokeCounts.txt') as f_in:
+        with open('Unihan_RadicalStrokeCounts_kRSKangXi.txt', "w+") as f_out:
+            for line in f_in:
+                if line[0] == '#' or len(line) <= 1:
+                    # Skip
+                    continue
+                elif line.split('\t')[1] == 'kRSKangXi':
+                    f_out.write(line)
+
+    # Mapping from radical number to radical string
+    # Radicals in Unicode range from '\u2f00' to '\u2fd6'
+    # See https://en.wikipedia.org/wiki/List_of_radicals_in_Unicode
+    number_to_radical = {}
+    for i in range(214):
+        number_to_radical[i+1] = chr(int('0x2F00', 16) + i)
+
+    # Get mapping from radical number to number of strokes in the radical
+    radical_numbers = [i+1 for i in range(214)]
+    stroke_counts = [1]*6 + [2]*23 + [3]*31 + [4]*34 + [5]*23 + [6]*29 + \
+                    [7]*20 + [8]*9 + [9]*11 + [10]*8 + [11]*6 + [12]*4 + \
+                    [13]*4 + [14]*2 + [15]*1 + [16]*2 + [17]*1
+    number_to_stoke_count = dict(zip(radical_numbers, stroke_counts))
+
+    # Each entry in CRS_list (code point radical stroke list) is:
+    # (code_point, character, radical_number, radical, radical_stroke_count,
+    # remaining_stroke_count)
+    CRS_list = []
```

Comment:
please use a better name, such as codepoint_radical_stroke_list instead of CRC_list.